### PR TITLE
Allow a --upload flag to upload benchmarks after running

### PR DESCRIPTION
**Pull request by jskeet**
Hard to really try this from where I am, but comments welcome...

### CODE REVIEW

File: build/runbenchmarks.sh [hunk 1]
```diff
@@ -2,10 +2,13 @@
 
 set -e
 
+declare -r ROOT=$(realpath $(dirname $0)/..)
+
 if [[ "$3" = "" ]]
 then
-  echo "Usage: runbenchmarks.sh <repo-directory> <results-directory> <target-framework> [target-framework...]"
-  echo "e.g. runbenchmarks.sj benchmark-tmp benchmark-results netcoreapp1.1 net45"
+  echo "Usage: runbenchmarks.sh <repo-directory> <results-directory> [--push] <target-framework> [target-framework...]"
```
Comment (malcolmr) [inline build/runbenchmarks.sh]: `--upload`, not `--push`.

File: build/runbenchmarks.sh [hunk 2]
```diff
@@ -2,10 +2,13 @@
...
+  echo "e.g. runbenchmarks.sj benchmark-tmp benchmark-results --upload netcoreapp1.1 net45"
```
Comment (malcolmr) [inline build/runbenchmarks.sh]: typo: `.sj` -> `.sh` here.

File: build/runbenchmarks.sh [hunk 3]
```diff
@@ -46,3 +57,12 @@ do
...
   shift
 done
+
+if [[ $UPLOAD == "true" ]]
+then
+  echo "Uploading benchmarks"
+  cd "$ROOT/build/BenchmarkUploader"
+  dotnet restore
+  dotnet run -- $RESULTS nodatime benchmarks/
+  echo "I would upload now"
```
Comment (malcolmr) [inline build/runbenchmarks.sh]: Unfinished?

**Comment by jskeet** [comment]
Whoops, thanks. All dumb mistakes fixed. Will start using this on the benchmark machines when I get home...

**Pull request merged**